sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql

```diff
@@ -0,0 +1,224 @@
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    metrics_ping_base
+  GROUP BY
+    client_id,
+    sample_id
+),
+_current AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+    first_reported_country,
+    first_reported_isp,
+    channel,
+    device_manufacturer,
+    device_model,
+    os_version,
+    app_version,
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    STRUCT(
+      IF(first_session.client_id IS NULL, FALSE, TRUE) AS is_reported_first_session_ping,
+      IF(metrics.client_id IS NULL, FALSE, TRUE) AS is_reported_metrics_ping,
+      CASE
+        WHEN first_session.adjust_info IS NOT NULL
+          THEN "first_session"
+        WHEN metrics.adjust_info IS NOT NULL
+          THEN "metrics"
+        ELSE NULL
+      END AS adjust_info__source_ping
+    ) AS metadata,
+    -- field to help us identify suspicious devices on iOS, for more info see: bug-1846554
+    (app_version = '107.2' AND submission_date >= '2023-02-01') AS is_suspicious_device_client,
+  FROM
+    first_seen
+  FULL OUTER JOIN
+    first_session_ping AS first_session
+  USING
+    (client_id, sample_id)
+  FULL OUTER JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id, sample_id)
+  WHERE
+    client_id IS NOT NULL
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    fenix_derived.firefox_android_clients_v2
+)
+SELECT
+  client_id,
+  sample_id,
+  COALESCE(_previous.first_seen_date, _current.first_seen_date) AS first_seen_date,
+  COALESCE(
+    _previous.first_reported_country,
+    _current.first_reported_country
+  ) AS first_reported_country,
+  COALESCE(_previous.first_reported_isp, _current.first_reported_isp) AS first_reported_isp,
+  COALESCE(_previous.channel, _current.channel) AS channel,
+  COALESCE(_previous.device_manufacturer, _current.device_manufacturer) AS device_manufacturer,
+  COALESCE(_previous.device_model, _current.device_model) AS device_model,
+  COALESCE(_previous.os_version, _current.os_version) AS os_version,
+  COALESCE(_previous.app_version, _current.app_version) AS app_version,
+  activations.is_activated,
+  -- below is to avoid mix and matching different adjust attributes
+  -- from different records. This way we always treat them as a single "unit"
+  IF(
+    _previous.adjust_ad_group IS NULL
+    AND _previous.adjust_campaign IS NULL
+    AND _previous.adjust_creative IS NULL
+    AND _previous.adjust_network IS NULL,
+    _current.adjust_info,
+    STRUCT(
+      _previous.submission_timestamp,
+      _previous.adjust_ad_group,
+      _previous.adjust_campaign,
+      _previous.adjust_creative,
+      _previous.adjust_network
+    )
+  ).*,
+  STRUCT(
+    COALESCE(
+      _previous.metadata.is_reported_first_session_ping
+      OR _current.metadata.is_reported_first_session_ping,
+      FALSE
+    ) AS is_reported_first_session_ping,
```

Comment:
current name on Android is just `reported_first_session_ping`. We can change it for consistency, but this field is used a lot so it'll break a bunch of queries once we update the view.

Reply:
ok, changed the naming for android.